tests/formatters.test.js

```diff
@@ -0,0 +1,78 @@
+import * as formatters from '../src/lib/formatters';
+
+test('serializeFilters method should return the updated search params', () => {
+  const testUrl = new URL('http://test.html?timeStamp=null&id=["dummyId123"]');
+  const searchParams = new URLSearchParams(testUrl.search.slice(1));
+
+  const extIds = ['addon1@test.com', 'addon2@test.com'];
+  const viewTypes = ['popup', undefined];
+  const types = ['api_call', 'content_script'];
+  const names = ['test@api'];
+  const tabId = 22;
+  const keyword = 'test@substring';
+  const timeStamp = { start: 1597686226302, stop: 1597686226302 };
+
+  const updateFilter = {
+    id: { exclude: new Set(extIds) },
+    viewType: { exclude: new Set(viewTypes) },
+    type: { exclude: new Set(types) },
+    name: { exclude: new Set(names) },
+    tabId,
+    keyword,
+    timeStamp,
+  };
+
+  const serializedFormat = formatters.serializeFilters(
+    searchParams,
+    updateFilter
+  );
+
+  expect(serializedFormat.get('id')).toEqual(JSON.stringify(extIds));
+  expect(serializedFormat.get('viewType')).toEqual(JSON.stringify(viewTypes));
```

Comment:
Hardcode the string instead of using `JSON.stringify`. It's not obvious from the code to those unfamiliar with the exact working of `JSON.stringify` that the stringified value is `["popup",null]`.

Reply:
Yeah, done :-)